pkg/storage/gcp/gcp.go

```diff
@@ -45,6 +45,14 @@ func NewWithCredentials(ctx context.Context, cred option.ClientOption) (*Storage
 	}, nil
 }
 
+func newWithBucket(bkt Bucket) *Storage {
+	return &Storage{
+		bucket:       bkt,
+		baseURI:      nil,
```

Comment:
We will have to test the BaseURL function so this shouldn't be nil. Maybe an argument to the newWithBucket function.

Reply:
agreed